File: python/exercise_88.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        pointer_nums1 = m-1
        pointer_nums2 = n-1
        pointer_current_index = m+n-1
        while pointer_nums2 >= 0:
            if pointer_nums1 >=0 and nums1[pointer_nums1] > nums2[pointer_nums2]:
                nums1[pointer_current_index] = nums1[pointer_nums1]
                pointer_nums1 = pointer_nums1-1
            else:
                nums1[pointer_current_index] = nums2[pointer_nums2]
                pointer_nums2 = pointer_nums2-1
            pointer_current_index = pointer_current_index-1

File: java/arrays/exercise_88.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int pointer1 = m-1;
        int pointer2 = n-1;
        for(int i = nums1.length-1;i>=0;i--){
            if(pointer1<0){
                nums1[i] = nums2[pointer2];
                pointer2--;
            }else{
                if(pointer2<0){
                    nums1[i] = nums1[pointer1];
                    pointer1--;                
                }else{
                    if(nums1[pointer1]>nums2[pointer2]){
                        nums1[i] = nums1[pointer1];
                        pointer1--;
                    }else{
                        nums1[i] = nums2[pointer2];
                        pointer2--;
                    }
                }
            }

        }

        return;
    }
}